File: src/nomanssave/bB.java
package nomanssave;

import javax.swing.ComboBoxModel;
import javax.swing.event.ListDataListener;

class bB implements ComboBoxModel {
  er eu;
  
  private bB(bl parambl) {}
  
  public int getSize() {
    return (bl.a(this.er) == null) ? 0 : (bl.a(this.er)).length;
  }
  
  public er v(int paramInt) {
    return (bl.a(this.er) == null) ? null : bl.a(this.er)[paramInt];
  }
  
  public void addListDataListener(ListDataListener paramListDataListener) {}
  
  public void removeListDataListener(ListDataListener paramListDataListener) {}
  
  public void setSelectedItem(Object paramObject) {
    this.eu = (er)paramObject;
    if (bl.b(this.er) >= 0) {
      er er1 = bl.c(this.er)[bl.b(this.er)].ar(0);
      if (this.eu != er1) {
        if (er1 != null) {
          int i = er1.aU().ordinal();
          int j = bl.c(this.er)[bl.b(this.er)].aq(i) - er1.aV();
          if (j < 0)
            j = 0; 
          bl.c(this.er)[bl.b(this.er)].e(i, j);
          bl.d(this.er)[i].setText(Integer.toString(j));
        } 
        if (this.eu == null) {
          bl.c(this.er)[bl.b(this.er)].a(0, null);
        } else {
          bl.c(this.er)[bl.b(this.er)].a(0, this.eu);
          int i = this.eu.aU().ordinal();
          int j = bl.c(this.er)[bl.b(this.er)].aq(i) + this.eu.aV();
          if (j < 0)
            j = 0; 
          bl.c(this.er)[bl.b(this.er)].e(i, j);
          bl.d(this.er)[i].setText(Integer.toString(j));
        } 
        bl.e(this.er).updateUI();
      } 
    } 
  }
  
  public Object getSelectedItem() {
    return this.eu;
  }
}


/* Location:              E:\NMSSE_Source\NMSSaveEditor.jar!\nomanssave\bB.class
 * Java compiler version: 8 (52.0)
 * JD-Core Version:       1.1.3
 */

File: NMSSaveEditor/nomanssave/mixed/bB.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Windows.Forms;
using System.Globalization;

namespace NMSSaveEditor
{

public class bB : ComboBoxModel {
   public er eu;
   // $FF: synthetic field
   public bl er;

   public bB(bl var1) {
      this.er = var1;
   }

   public int getSize() {
      return bl.a(this.er) == null ? 0 : bl.a(this.er).Length;
   }

   public er v(int var1) {
      return bl.a(this.er) == null ? null : bl.a(this.er)[var1];
   }

   public void addListDataListener(ListDataListener var1) {
   }

   public void removeListDataListener(ListDataListener var1) {
   }

   public void setSelectedItem(Object var1) {
      this.eu = (er)var1;
      if (bl.b(this.er) >= 0) {
         er var2 = bl.c(this.er)[bl.b(this.er)].ar(0);
         if (this.eu != var2) {
            int var3;
            int var4;
            if (var2 != null) {
               var3 = var2.aU().ordinal();
               var4 = bl.c(this.er)[bl.b(this.er)].aq(var3) - var2.aV();
               if (var4 < 0) {
                  var4 = 0;
               }
                bl.c(this.er)[bl.b(this.er)].e(var3, var4);
               bl.d(this.er)[var3].SetText(Convert.ToString(var4));
            }
             if (this.eu == null) {
               bl.c(this.er)[bl.b(this.er)].a(0, (er)null);
            } else {
               bl.c(this.er)[bl.b(this.er)].a(0, this.eu);
               var3 = this.eu.aU().ordinal();
               var4 = bl.c(this.er)[bl.b(this.er)].aq(var3) + this.eu.aV();
               if (var4 < 0) {
                  var4 = 0;
               }
                bl.c(this.er)[bl.b(this.er)].e(var3, var4);
               bl.d(this.er)[var3].SetText(Convert.ToString(var4));
            }
             bl.e(this.er).updateUI();
         }
      }
    }

   public Object getSelectedItem() {
      return this.eu;
   }

   // $FF: synthetic method
   public Object getElementAt(int var1) {
      return this.v(var1);
   }

   // $FF: synthetic method
   public bB(bl var1, bB var2) {
      // Constructor chain: base(var1)
   }

}


}